src/lib.rs

```diff
@@ -222,6 +222,8 @@ pub struct Shape {
     // Indentation + any already emitted text on the first line of the current
     // statement.
     pub offset: usize,
+    // Alignment to other structures
+    pub alignment: usize,
```

Comment:
Rather than add this to `Shape`, can you pass this as an extra arg to the relevant functions?